### PR TITLE
feat: adds useTokenFees hook

### PR DESCRIPTION
**Pull request by gunnmgirl**
## PR-Codex overview
This PR adds new hooks and types related to token fees handling. It also refactors existing hooks to use `PartialOptions` instead of `PartialBuildOptions`.

### Detailed summary
- Added `TokenFeesHookArgs` type
- Added `useTokenFees` hook
- Refactored existing hooks to use `PartialOptions`
- Updated `utils` functions to use `PartialOptions`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Review by joepegler — COMMENTED** [review]
There is a smartAccount.getTokenFees method I think

File: src/stories/components/TokenFees.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    manyOrOneTransactions: {
+      to,
+      value,
+    },
+    buildUseropDto: {
+      paymasterServiceData: { mode: paymasterMode },
+    },
+  });
+
+  console.log("data-", data);
```
Comment (malcodeman) [inline src/stories/components/TokenFees.tsx:18]: Needless console log

Comment (gunnmgirl) [reply src/stories/components/TokenFees.tsx]: Removed 👍

**Comment by malcodeman** [comment]
@gunnmgirl please resolve conflicts and we can merge 👍

**Review by joepegler — APPROVED** [review]
lgtm, nice job

**Pull request merged**